src/events.js

```diff
@@ -70,6 +81,51 @@ module.exports = function(ctx) {
     currentMode.mouseout(event);
   };
 
+  events.touchstart = function(event) {
+    // Prevent emulated mouse events because we will fully handle the touch here.
+    // This does not stop the touch events from propogating to mapbox though.
+    event.originalEvent.preventDefault();
+    if (!ctx.options.touchEnabled) {
+      return;
+    }
+
+    touchStartInfo = {
+      time: new Date().getTime(),
+      point: event.point
+    };
+    var target = featuresAt.touch(event, null, ctx)[0];
+    event.featureTarget = target;
+    currentMode.touchstart(event);
```

Comment:
I wrestled with this in my head. It's obviously a DRY-explicitness tradeoff. My personal preference is to make things explicit when the interaction mechanism is fairly different in order to make changes easier later.

But that's just my opinion. If you think it would be better to have handlers with one code path then I'll go ahead and change them?